src/openapi/streaming/subscription.js

```diff
@@ -218,13 +228,20 @@ function onSubscribeError(referenceId, response) {
         return;
     }
 
-    this.currentState = STATE_UNSUBSCRIBED;
-    log.error(LOG_AREA, 'An error occurred subscribing', { response, url: this.url });
+    if (response && response.status === '504' && this.retryCounter++ < MAX_RETRIES) {
+        log.error(LOG_AREA, 'Received gateway timeout when subscribing. Retrying.', { response, url: this.url });
 
-    // if we are unsubscribed, do not fire the error handler
-    if (this.queue.peek() !== ACTION_UNSUBSCRIBE) {
-        if (this.onError) {
-            this.onError(response);
+        this.currentState = STATE_SUBSCRIBED; // set to subscribed so that unsubscribe request isn't ignored

```

Comment:
Ideally I think we should peek next action, and check if it's unsubscribe. If so, no need for retrying.